### PR TITLE
fix: Override block.basefee call in ZK

### PR DESCRIPTION
**Pull request by Jrigada**
## Motivation
`block.basefee` was different in non-zk execution than the one on zk execution when executing tests. 

## Solution
Override base fee with the base fee coming from EVM.

### CODE REVIEW

**Review by Karrq — APPROVED** [review]

File: crates/zksync/core/src/vm/tracer.rs [hunk 1]
```diff
@@ -56,6 +56,9 @@ const SELECTOR_SYSTEM_CONTEXT_BLOCK_NUMBER: [u8; 4] = hex!("42cbb15c");
 /// Selector for `getBlockTimestamp()`
 const SELECTOR_SYSTEM_CONTEXT_BLOCK_TIMESTAMP: [u8; 4] = hex!("796b89b9");
 
+/// Selector for `block.baseFee`
+const SELECTOR_BLOCK_BASE_FEE: [u8; 4] = hex!("6ef25c3a");
```
Comment (nbaztec) [inline crates/zksync/core/src/vm/tracer.rs:60]: The docs need to follow the above format. Also the selector is for `baseFee()` function

**Review by nbaztec — APPROVED** [review]

File: crates/zksync/core/src/vm/runner.rs [hunk 2]
```diff
@@ -253,6 +257,8 @@ where
         },
         block_number: env.block.number,
         block_timestamp: env.block.timestamp,
+        block_basefee: env.block.basefee,
```
Comment (nbaztec) [inline crates/zksync/core/src/vm/runner.rs:260]: Please set this to `max(max_fee_per_gas, env.block.basefee)` already here

File: crates/zksync/core/src/vm/tracer.rs [hunk 3]
```diff
@@ -259,6 +269,23 @@ impl<S: Send, H: HistoryMode> DynTracer<S, SimpleMemory<H>> for CheatcodeTracer
...
+            let calldata = get_calldata(&state, memory);
+            let current = state.vm_local_state.callstack.current;
+
+            if current.code_address == SYSTEM_CONTEXT_ADDRESS &&
+                calldata.starts_with(&SELECTOR_BASE_FEE)
+            {
+                let mut ret = self.call_context.block_basefee;
+                if self.call_context.block_basefee > self.call_context.max_fee_per_gas {
+                    ret = self.call_context.max_fee_per_gas;
+                }
```
Comment (nbaztec) [inline crates/zksync/core/src/vm/tracer.rs:283]: This check can be directly done at input

File: crates/zksync/core/src/vm/tracer.rs [hunk 4]
```diff
@@ -98,10 +98,9 @@ pub struct CallContext {
     pub block_number: rU256,
     /// The current block timestamp
     pub block_timestamp: rU256,
-    /// block base fee
+    /// block base fee (if base fee is higher than max fee per gas, it will be overridden to max
+    /// fee per gas)
```
Comment (nbaztec) [inline crates/zksync/core/src/vm/tracer.rs:102]:
```suggestion
    /// The current block basefee
```

**Pull request merged**